autogpt/setup.py

```diff
@@ -75,3 +134,64 @@ def prompt_user() -> AIConfig:
         ]
 
     return AIConfig(ai_name, ai_role, ai_goals)
+
+
+def generate_aiconfig_automatic(user_prompt) -> AIConfig:
+    """Generates an AIConfig object from the given string.
+
+    Returns:
+    AIConfig: The AIConfig object tailored to the user's input
+    """
+
+    system_prompt = """
+Your task is to devise up to 5 highly effective goals and an appropriate role-based name (_GPT) for an autonomous agent, ensuring that the goals are optimally aligned with the successful completion of its assigned task.
+
+The user will provide the task, you will provide only the output in the exact format specified below with no explanation or conversation.
+
+Example input:
+Help me with marketing my business
+
+Example output:
+Name: CMOGPT
+Description: a professional digital marketer AI that assists Solopreneurs in growing their businesses by providing world-class expertise in solving marketing problems for SaaS, content products, agencies, and more.
+Goals:
+- Engage in effective problem-solving, prioritization, planning, and supporting execution to address your marketing needs as your virtual Chief Marketing Officer.
+
+- Provide specific, actionable, and concise advice to help you make informed decisions without the use of platitudes or overly wordy explanations.
+
+- Identify and prioritize quick wins and cost-effective campaigns that maximize results with minimal time and budget investment.
+
+- Proactively take the lead in guiding you and offering suggestions when faced with unclear information or uncertainty to ensure your marketing strategy remains on track.
+"""
+
+    # Call LLM with the string as user input
+    messages = [
+        {
+            "role": "system",
+            "content": system_prompt,
+        },
+        {
+            "role": "user",
+            "content": f"Task: '{user_prompt}'\nRespond only with the output in the exact format specified in the system prompt, with no explanation or conversation.\n",
+        },
+    ]
+    output = create_chat_completion(messages, CFG.fast_llm_model)
+
+    # Debug LLM Output
+    if CFG.debug_mode:
+        logger.debug(f"AI Config Generator Raw Output: {output}")
```

Comment:
Assuming the logger is set up properly, the `if` isn't necessary and the logger will only output messages >= its configured log level. `debug_mode` should ideally only be used to define that log level.

Reply:
Ah, that's a great system if it is indeed set up that way, thanks for letting me know :)